### ISSUE TITLE
Which parameters get tuned?

### ISSUE THREAD

**Issue by Dboingue** [Dboingue/Hello-Stockfish#2]
Which parameters get tuned differently now that NNue is in the picture?  There was related issue in origin. about neglecting the classical evaluation by not autotuning (a.k.a. afaik, global optimization, too vague, it could include training, which autotuning is not careful about generalization and overfitting (and possibly optimizing toward the test set sampling biases, and getting increasing better at it, but only for the test set, but for a more general chess challenge for example).  Maybe this is not the case anymore. but i have not read explicit confirmation that the auto-tuning in  SF, is doing some cross-validation in the fishnet pipeline.. if that is where all of it happens.

- [ ] gathers fishtest and tuning code links here
- [ ] find which parameters in static eval are being from human knowledge, or tuned.  e.g. material count is in design., positional parameters are not, as there is only qualtiative feature human knowledge, (0,1 being the trivial ordering, quant?).
- [ ]  (i.e. drilling down the variable in the source code of SF, until variable is assigned a constant or a valued function output.   once mapping between dev version and exec version is available, centralize all parameters that represent the information of the the classical eval.
- [ ] same for tuning
- [ ] first measure if Sisiphean ambitiion

_Originally posted by @Dboingue in https://github.com/Dboingue/Hello-Stockfish/issues/1#issuecomment-885249595_

**Comment by Dboingue** [comment]
This issue is now about the whole of SF.  as our source code searching and reading in the heuristic evaluation (classical SF) parts show many hardwired values in the score impacting assignment code lines.  Factors with no names.  Hence, not possible to know from stockfish code itself, if the design of those value is part of tuning. or inherited from past versions.  Or some mixture.

Some parameters are, per fishtest wiki being tuned with modern global optimization methods.  But the parameter set is not defined there, neither the formulation of the objective function, which is a basic component of any  such optimization problem, as far as I know.  

The objective function description can be high level, without enumrating each parameters.  If some vectorization of all parameters keeping some meaningful categorization with respect to which conceptual part of the code is affected, then a matrix formulation without explicit index within line can be written (but that might required or benefit from a high level agreed description of what SF is doing).  And its component sub-matrices to substitute, themselves written from high level.  Making the mathematical formulation accessible sequentially. in a text file (left to right, top to bottom, that is how we are forced to communicate).  The web guide has done a lot of work on the static part of SF, but we also need the most important part, what is the search doing with such static function of input position FEN exactly (in highest level terms).

- [ ] find my stockfish issue asking for  a list of all parameters, constant or tuned (and source code reference).
